_examples/blockchain.adoc

```diff
@@ -0,0 +1,107 @@
+Installing our notebook is very simple:  we just need to create an OpenShift project and install and run the notebook application.
+Make sure you're logged in to OpenShift and have selected the right project, and then execute the following two commands:
+
+....
+oc new-app jkremser/bitcoin-notebook:tutorial-1.0.0
+oc expose svc/bitcoin-notebook
+....
+
+for the Jupyter notebook and
+
+....
+oc new-app jkremser/bitcoin-spark-notebook:tutorial-1.0.0
+oc expose svc/bitcoin-spark-notebook
+....
+
+Once both routes are exposed the both notebooks should be available. You
+can find the URLs if you run `oc get routes`.
+It is also possible for the Jupyter notebook to set
+environmental variable that assures it will connect to an external Spark cluster.
+
+To do that you can optionally:
+`oc new-app jkremser/bitcoin-notebook:tutorial-1.0.0 -e SPARK_MASTER=spark://c:7077`
+
+Where `c` is the hostname of the Spark cluster.
+
+[[usage]]
+== Usage
+Both notebooks comes with some prepared data in the parquet files, but part of the
+project is also the https://github.com/Jiri-Kremser/bitcoin-insights/tree/master/parquet-converter[parquet-converter]
+that takes the raw Blockchain data and converts
+it to the format that is digestible by Spark. It assumes the official bitcoin wallet to be installed
+that itself downloads the whole Blockchain (~150 GB). However, it works also with
+partially downloaded Blockchain.
+
+* Jupyter Notebook (GraphFrames and NetworkX)
++
+Interacting with the Jupyter notebook is very simple. If you select a cell with
+Python code you can 1) edit it or 2) execute it by pressing the "run cell" button
+in the toolbar or by pressing shift + enter. This notebook includes cells with
+instructions for running the program as well as the explanatory video about how the Blockchain internally works
++
+The description of what the notebook actually does should be part of the notebook itself.
+In short this notebook demonstrates the degree distribution of the bitcoin addresses
+and compares with older Blockchain data. It also visualizes sub-graphs using sigma.js library and
+with https://networkx.github.io/[NetworkX] python library.
+
+* Spark Notebook (GraphX and PageRank)
++
+While the previous notebook was trying to look at the graph as a whole, find
+some common properties and compare some statistics how they developed in time,
+this one demonstrates the famous PageRank algorithms and finds outliers in the graph and
+sort them by the significancy.
+
+[[expansion]]
+== Expansion
+Currently the number of edges in the graph is relatively large and it is due to
+the fact that during the conversion to parquet files we do following. For each
+transaction in each block we create an edge between each input in the transaction
+and each output in the transaction. This means that in case of multi-input (N),
+multi-output (M) transaction we end up with N times M edges.
+
+Useful improvement would be representing the concept of transaction in the resulting
```

Comment:
s/Useful improvement/A useful improvement/

s/of transaction/of a transaction/